utils/utils.go

```diff
@@ -119,34 +120,42 @@ func DecodeProof(encodedProofString string) (*types.Proof, error) {
 		return nil, err
 	}
 
+	if len(encodedProof) < 16 {
+		return nil, fmt.Errorf("encoded proof is too short")
```

Comment:
Could you check if a proof exists before throwing this error? Just to have a cleaner return log